### DISCUSSION TITLE
Particle out

### DISCUSSION THREAD

**Discussion by sddadsadsadsddsd** [DualSPHysics/DualSPHysics#224] [DUALSPHYSICS_V5.2]
How to solve the problem of particle out,Please give me some advice
![捕获](https://github.com/user-attachments/assets/74ea9aa7-cfcb-4fa5-9796-412dd1a35a3e)

**Comment by AhmedSalih3d** [comment]
Particle out is not necessary a problem, depending on what you simulate.

In your case it is, since the particles are leaving the box you have defined. One way to solve particle out would be to add additional boundary layers, instead of 1 you have right now, try adding 2 more, so 3 in total. Do this by widening the "thickness" of the box domain outwards.

An easy way to do this is using the `layers` option for `drawbox` in the XML, please consult the documentation/examples.

Kind regards

**Comment by alexbexe** [comment]
You will need to share the XML with us to check what is going wrong.
It can be related with boundary conditions, compressibility or the time step

Regards

**Reply by 258458** [reply]
> 您需要与我们共享 XML 以检查出了什么问题。它可以与边界条件、可压缩性或时间步长有关
> 
> 问候
Hello, I would like to ask where the setting for compressibility is reflected in the code?
